src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx

```diff
@@ -113,6 +113,224 @@ t8_geometry_squared_disk::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreei
   }
 }
 
+/**
+ * Map the faces of an oktaeder to a spherical surface.
+ * \param [in]  cmesh      The cmesh in which the point lies.
+ * \param [in]  gtreeid    The global tree (of the cmesh) in which the reference point is.
+ * \param [in]  ref_coords  Array of \a dimension many entries, specifying a point in [0,1]^dimension.
+ * \param [out] out_coords  The mapped coordinates in physical space of \a ref_coords.
+ */
+void
+t8_geometry_triangulated_spherical_surface::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreeid,
+                                                              const double *ref_coords, const size_t num_coords,
+                                                              double *out_coords) const
+{
+  double n[3]; /* Normal vector of the current triangle. */
+  double r[3]; /* Radial vector through one of triangle's corners. */
+  double p[3]; /* Position vector on the triangle plane. */
+
+  t8_locidx_t ltreeid = t8_cmesh_get_local_id (cmesh, gtreeid);
+  double *tree_vertices = t8_cmesh_get_tree_vertices (cmesh, ltreeid);
+
+  n[0] = tree_vertices[0] + tree_vertices[3] + tree_vertices[6];
+  n[1] = tree_vertices[1] + tree_vertices[4] + tree_vertices[7];
+  n[2] = tree_vertices[2] + tree_vertices[5] + tree_vertices[8];
+
+  {
+    /* Normalize vector `n`. */
+    const double norm = sqrt (n[0] * n[0] + n[1] * n[1] + n[2] * n[2]);
+    n[0] = n[0] / norm;
+    n[1] = n[1] / norm;
+    n[2] = n[2] / norm;
+  }
+
+  r[0] = tree_vertices[0];
+  r[1] = tree_vertices[1];
+  r[2] = tree_vertices[2];
+
+  {
+    /* Normalize vector `r`. */
+    const double norm = sqrt (r[0] * r[0] + r[1] * r[1] + r[2] * r[2]);
+    r[0] = r[0] / norm;
+    r[1] = r[1] / norm;
+    r[2] = r[2] / norm;
+  }
+
+  /* Init output coordinates with zeros. */
+  for (size_t i_coord = 0; i_coord < num_coords; i_coord++) {
```

Comment:
I don't think you need to zero-initialize, as in line 221 - 223 you can directly set the out_coords to R*p[0]

Reply:
Done.